src/main/java/com/salesforce/dynamodbv2/mt/mappers/MtAmazonDynamoDbStreamsBase.java

```diff
@@ -143,32 +136,18 @@ public GetRecordsResult getRecords(GetRecordsRequest getRecordsRequest) {
             LOG.debug("getRecords request={}", getRecordsRequest);
         }
 
-        ShardIterator iterator = ShardIterator.fromString(getRecordsRequest.getShardIterator());
-        String arn = iterator.getArn();
-        StreamArn streamArn = parse(arn);
+        final ShardIterator iterator = ShardIterator.fromString(getRecordsRequest.getShardIterator());
+        final String arn = iterator.getArn();
+        final StreamArn streamArn = parse(arn);
 
         // transform tenant-aware into DynamoDB iterator request
-        GetRecordsRequest request = getRecordsRequest.clone()
-            .withShardIterator(iterator.withArn(streamArn.toDynamoDbArn()).toString());
-
-        // create tenant record mapper and filter
-        Function<Record, MtRecord> recordMapper = getRecordMapper(streamArn);
-        Predicate<MtRecord> recordFilter = getRecordFilter(streamArn);
-
-        // perform actual lookup
-        GetRecordsResult result = getRecords(request, recordMapper, recordFilter);
-
-        if (!(result instanceof MtGetRecordsResult)) {
-            result = new MtGetRecordsResult()
-                .withRecords(result.getRecords())
-                .withNextShardIterator(result.getNextShardIterator())
-                .withLastSequenceNumber(Optional.ofNullable(result.getRecords())
-                    .filter(not(List::isEmpty))
-                    .map(Iterables::getLast)
-                    .map(Record::getDynamodb)
-                    .map(StreamRecord::getSequenceNumber)
-                    .orElse(null));
-        }
+        final GetRecordsRequest request = new GetRecordsRequest()
+            .withShardIterator(iterator.withArn(streamArn.toDynamoDbArn()).toString())
+            .withLimit(getRecordsRequest.getLimit());
+
+        final MtGetRecordsResult result = streamArn instanceof MtStreamArn
+            ? getMtRecords(request, (MtStreamArn) streamArn)
+            : getMtRecords(request, streamArn);
```

Comment:
Depending on parse result, dispatches to one of two methods implemented by subclasses.

Reply:
Do we want to have a flag that gets passed into the `MtAmazonDynamoDbStreamsBase` that tells it that it should require context?  This would preclude accidentally falling back to reading all records from the stream accidently.